git_greppers/github_handler.sh

```
@@ -0,0 +1,120 @@
+  pr_mr_number=$(echo "$commit_msg" | grep -o -E "(Merge pull request|PR) #[0-9]+" | grep -o -E "#[0-9]+" | head -1)
+  
+  debug 3 "GitHub PR pattern search result: '$pr_mr_number'" >&2
+  
+  if [[ -n $pr_mr_number ]]; then
+    # Found a PR reference in the commit message
+    pr_mr_info="GitHub PR $pr_mr_number (https://github.com/$repo_identifier/pull/${pr_mr_number#\#})"
+    debug 2 "Found PR reference in commit message: $pr_mr_info" >&2
+  elif command -v gh &> /dev/null; then
+    debug 2 "GitHub CLI available, attempting to use it to find PR info" >&2
+    # If GitHub CLI is available, try to use it to find PR information
+    local gh_pr=$(gh pr list --search "$commit_hash" --json number,title,url --limit 1 2>/dev/null)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)